### ISSUE TITLE
Feature Request: {Trinkets Integration}

### ISSUE THREAD

**Issue by Dumpling00** [Lightman314/LightmansCurrencyFabric#28]
Hello! This is a bit of a situational request, but since Curios API isn't available on Fabric 1.20.1, would it be possible to make the wallet compatible with a trinkets slot? Maybe the leggings one so that it can still show attached to the belt?

The reason I ask this is because I'm running a bit of an adventure game with Switchy, where each player can have different character presets, and through some of its modules, it's possible to make things like health, inventory, exp, trinkets etc. unique to each character preset.
Right now, because the wallet is its own thing, each preset shares the same wallet. I'd like to make it so each character can have their own wallet.

Thank you for your consideration!

**Comment by Dumpling00** [comment]
Sorry to bother but any chance this can be looked at in the near future? I'm holding off on starting an adventure campaign with friends and this is the perfect mod to go with.

**Comment by Lightman314** [comment]
Just out of curiosity, why would each character preset need a different wallet?

**Comment by Dumpling00** [comment]
> Just out of curiosity, why would each character preset need a different wallet?

Again, I admit this is very situational, but this is treated as a roleplay adventure, where each character takes up a role and goes off exploring one part / dimension of the world. This is also so that, in the event of streaming, the player can focus on a given adventure with a given character and not have access to the entirety of the money that's been collectively collected from all the other presets.
Example, if I start an adventure with Character B, I wouldn't want them to have a thousand coins already on them. And if at any given point I want/need to switch to Character A, in the middle of another adventure, I wouldn't want them to have a different amount of coins in their wallet just because i spent/collected some with a different character.

Hope this makes it clearer. If not, please don't hesitate to ask more questions.

**Comment by Lightman314** [comment]
Interesting and unique idea. You're basically just playing multiple characters and using some method of switching between them is my understanding of it?

If making wallets compatible with the trinkets slot is all it'll take to make this work for you, I'm down for making it compatible, though the earliest I'll probably get to work on it will probably be this weekend.

**Comment by Dumpling00** [comment]
> Interesting and unique idea. You're basically just playing multiple characters and using some method of switching between them is my understanding of it?
> 
> If making wallets compatible with the trinkets slot is all it'll take to make this work for you, I'm down for making it compatible, though the earliest I'll probably get to work on it will probably be this weekend.

That's exactly right! Really all I'd need is the wallet to be placeable in the trinket slot attached to the leggings (ideally the one seen below), and Switchy will do the rest! If you could, that would be incredible.

Although, even by moving the phisical wallet to the trinket's slot, they might still share the same "bank account" in case any money is moved into the ATM system, but that's a non-issue since we're not planning to use the ATM feature. Worth mentioning we'll be playing through Essentials' peer-to-peer hosting feature so this wouldn't be on an actual server.

If it's not too much trouble, maybe an option that allows us to turn the bank account feature off could solve everything? Mostly as an anti-cheat to avoid other players setting up an acount in secret and sharing money across all the other characters). We're running a bit of a medieval/steampunk themed adventure, so we're not gonna need that kind of interconnectivity for long-distance payments.

Trinket Slot (If it doesn't normally show up, worth mentioning the highlighted one might be added by Not Enough Trinket Slots)
![image](https://github.com/Lightman314/LightmansCurrencyFabric/assets/44829013/cd1593b4-cc40-4018-b1b9-5044a827adf8)

Additionally, if at all possible and if it's not much trouble, a way to disable emerald, diamond and netherite tier coins, so that we'd only get copper, iron and gold coins to work with. This is because we believe this could finally give gold a proper value and use, reserving diamond and netherite for tools and equipment only, while keeping the emeralds exclusive to Villagers Trades.

So for full disclorure, I absolutely love the ore idea here with coins, the way you can stack them and place them, and the whole vending machine aspect, but it starts to become too complex/intricate for me and the needs of the campaign when introducing bank accounts, ATMs, tickets etc. so I would really love a way to make it more "Lite".
I know this is probably asking too much, but I thinkt it could also make it accessible to more casual, D&D-like playstyles.

Once again thank you so much for your time and consideration.

**Comment by Dumpling00** [comment]
Hey just wanna say sorry if this is too high scope. It'd be amazing, sure, but I'd be perfectly happy with just the trinket slot integration, just don't wanna put any unwanted pressure on you!

**Comment by Lightman314** [comment]
Got the trinkets compatibility working, so this should *hopefully* be released soon:
![image](https://github.com/Lightman314/LightmansCurrencyFabric/assets/4554273/2447e77e-b26c-4007-999b-f5ad4150aae1)


At the present moment there's no official way to disable bank accounts, nor do I ever really plan on adding any method of doing so, but you can somewhat do it anyways by disabling the ATM recipe via other mods (I normally recommend CraftTweaker for this, and it does appear to have a fabric version). You can then disable the netherite wallets bank account access by editing the server config's `wallet.bankLevel` option to 6 (the server config options can be  found in the `config/lightmanscurrency-server.lcconfig` file)

As for disabling coins, you can remove any unwanted coins from in the MasterCoinList config (found in `config/lightmanscurrency/MasterCoinList.json`, full tutorial can be found [here](https://github.com/Lightman314/LightmansCurrency/wiki/Coin-List-Config-(Pre%E2%80%902.2))), simply just remove any undesired coins from there and it'll remove them from the coin list so that they're no longer treated as coins (can't be exchanged into smaller coins, smaller coins won't be turned into them, etc.).
You can stop mobs & chests from dropping/spawning the smaller coins by either setting the T4-T6 `lootItem` options in the common config (`config/lightmanscurrency-common.lcconfig`) to a blank input ("" or just `lootItemTX=`), or by moving all entity/chest loot tables from their T4-T6 lists into T1-T3 lists.

**Comment by Lightman314** [comment]
Added in LC Fabric v1.0.2.3 for 1.20.1. Will eventually be backported to 1.19.2, but I'm tired rn, so it'll be a day or two most likely.

**Comment by Dumpling00** [comment]
> Added in LC Fabric v1.0.2.3 for 1.20.1. Will eventually be backported to 1.19.2, but I'm tired rn, so it'll be a day or two most likely.

Hey there thank you for your time-
Unfortunately the new version seems unstable now for some reason. I've tried multiple times (uninstalling the mod, reinstalling it, deleting the config files, clearing the cache etc.

Everything works perfectly without LC, but as soon as I add it, I can maybe spend 40 seconds in my world before Minecraft force-quits. No crash report, no nothing. The LiveLog also doesn't seem to give me anything telling exept for an occasional "Can't keep up! Is the server overloaded? Running xms or x ticks behind" when loading new chunks.

I have made a backup of the world prior to installing this (as I always do before installing new mods), and it almost feels like the world got corrupted when LC was installed, because the crash keeps happening even after I delete every trace of the mod, so long as I keep playing on that particular copy of the world.

I'm really not sure how to help here.
